### PR TITLE
[DOCS] Adds machine learning node type and filters

### LINKED ISSUE #49099: Document machine learning "l" node.role
**Description of the problem including expected versus actual behavior**:
[https://www.elastic.co/guide/en/elasticsearch/reference/7.4/cat-nodes.html](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/cat-nodes.html) contains : 
```
node.role, r, role, nodeRole
(Default) Roles of the node. Returned values include m (master-eligible node), d (data node), i (ingest node), v (voting-only node), and - (coordinating node only).

For example, mdi indicates a master-eligible data and ingest node.
```

However it is missing machine learning node added in PR [45174](https://github.com/elastic/elasticsearch/pull/45174/files#diff-0063bb7d32496101a51e81d104f91dba) -> it looks like the plan was to add it to the doc at the same time as voting-only node but this did not make it in [48162](https://github.com/elastic/elasticsearch/pull/48162)

### PR DESCRIPTION
**Pull request by lcawl**
Fixes https://github.com/elastic/elasticsearch/issues/49099

Preview:
http://elasticsearch_49121.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/cluster.html
http://elasticsearch_49121.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/cat-nodes.html#cat-nodes-api-query-params

### CODE REVIEW

**Comment by lcawl** [comment]
@elasticmachine run elasticsearch-ci/docs

File: docs/reference/cat/nodes.asciidoc [hunk 1]
```diff
@@ -43,9 +43,11 @@ Valid columns are:
 
 `node.role`, `r`, `role`, `nodeRole`::
 (Default) Roles of the node. Returned values include `m` (master-eligible node),
-`d` (data node), `i` (ingest node), `v` (voting-only node), and `-` (coordinating node only).
+`d` (data node), `i` (ingest node), `l` (machine learning node), `v`
+(voting-only node), and `-` (coordinating node only).
 +
-For example, `mdi` indicates a master-eligible data and ingest node.
+For example, `mdi` indicates a master-eligible data and ingest node. See
```
Comment (droberts195) [inline docs/reference/cat/nodes.asciidoc]: I think the letters are now sorted alphabetically, so `mdi` would be shown as `dim`.

Comment (lcawl) [reply docs/reference/cat/nodes.asciidoc]: You're right.  I was hoping it would say "mild", but no such luck :)

**Review by droberts195 — APPROVED** [review]
LGTM

**Pull request merged**